racket/collects/racket/contract/region.rkt

```diff
@@ -477,21 +477,24 @@
                                           (syntax/loc stx
                                             (begin
                                               (define-values () (begin auto-check ... (values)))
-                                              (define (guard super-field ... non-auto-name ... struct-name)
+                                              (define (guard-contract super-field ... non-auto-name ...)
```

Comment:
Is there now any difference between this `guard-contract` and `values`?

The only differences I'm aware of are arity and name. The arity shouldn't matter here. The name shouldn't matter either, as the error messages come from the `contract` form which supplies the name with `(quote maker)`.

So you should be able to get by without defining `guard-contract`, with
```
(define (guard super-field ... non-auto-name ... struct-name)
  ((contract (-> super-contract ... non-auto-contracts ... any)
             values
             ....)
   super-field ... non-auto-name ...))
```

One more minor question, I noticed you replaced `(current-contract-region)` with `(quote maker)`. I don't think there's good reason to change it. Since it's a syntax parameter not a run-time parameter, it won't be affected by being inside the function body.

Reply:
Yes, made these changes, but this will probably change anyway with the new `->`